### ISSUE TITLE
choosing local switch location?

### ISSUE THREAD

**Issue by progman1** [ocaml/opam#5653]
having a local development copy of the compiler
I would like to make more than one local switch (for cross-compilers from it.
however I have found the only invocation of opam that installs a local switch
from a local copy of the ocaml sources is
```
opam switch create .
```
which puts _opam in the same directory as the sources.
but this means only one _opam can be created.
I see that when a  compiler is chosen for download from the official 
opam repo, a _opam directory can be chosen:
```
opam switch create ../arm64   4.14.0
```
and thus any number of  switches can be had.

am I mistaken in believing multiple local switches cannot be made from a local copy of sources?

**Comment by progman1** [comment]
I think I have a workaround but may have uncovered a possible bug.
```
opam sw create <the directory for housing _opam> --empty
opam install <path to local ocaml sources> --working-dir --switch=<_opam path>
```
...and --working-dir seemingly has no effect (other than no git/dir-content difference announcement) and the source used is coming from the last commit.

**Comment by rjbou** [comment]
Indeed, you can only have a single local opam switch (`_opam` directory) per directory, so per local copy of sources. You're workaround is the good way to do it.

On working dir issue, is the package pinned before? 
Can you share the output of the install command with `-vv` and `--debug`.

**Comment by rjbou** [comment]
Any news on this issue ?

**Comment by kit-ty-kate** [comment]
> I think I have a workaround but may have uncovered a possible bug.
> 
> ```
> opam sw create <the directory for housing _opam> --empty
> opam install <path to local ocaml sources> --working-dir --switch=<_opam path>
> ```
> 
> ...and --working-dir seemingly has no effect (other than no git/dir-content difference announcement) and the source used is coming from the last commit.

I believe this was fixed in 2.4 by https://github.com/ocaml/opam/pull/6209.

I'm leaving this ticket open for the feature request (multiple local switches in a directory) that i've seen being asked for elsewhere a couple of times.
